meshnets/utils/model_training.py

```diff
@@ -62,12 +61,59 @@ def compute_train_stats(dataset, max_examples=1000):
     }
 
 
+def make_dataloader(version,
+                    split_percentage,
+                    split_start,
+                    num_proc,
+                    writer_batch_size,
+                    batch_size,
+                    num_workers_loader,
+                    shuffle=True):
+    """Make a dataloader for the wind tunnel dataset."""
+    if split_start == 'beginning':
+        split = f'train[:{split_percentage:.0%}]'
+    elif split_start == 'end':
+        split = f'train[-{split_percentage:.0%}:]'
+    else:
+        raise ValueError(f'Invalid split_start: {split_start}')
+    dataset = datasets.load_dataset('inductiva/wind_tunnel',
+                                    version=version,
+                                    split=split,
+                                    download_mode='force_redownload')
+    dataset = dataset.map(
```

Comment:
For simplicity purposes these next three lines which process the data to create the desired features and structure, should be in a separate method. Wdyt?

Reply:
Because this whole function could be also used for other datasets, with different dataset-processing methods.

Reply:
Not sure I follow. The `map` is a general method of hugging face datasets. But the processing function being used are already problem specific. From datasets following the structure chosen for our HuggingFace dataset they create edge_features, node_features, ...